src/frontend/classic/components/MangaViewer.svelte

```diff
@@ -0,0 +1,217 @@
+            imageWidth = maxImageWidth;
+        } else if (winWidth > carbonMaxBreakpoint) {
+            imageWidth = minImageWidth;
+        } else {
+            // smooth transition from 100% to 50%
+            imageWidth =
+                ((maxImageWidth - minImageWidth) /
+                    (carbonLgBreakpoint - carbonMaxBreakpoint)) *
+                    (winWidth - carbonLgBreakpoint) +
+                maxImageWidth;
+        }
+    };
+
+    const getPromiseImages = (
+        entries: IMediaChild[]
+    ): Promise<HTMLImageElement>[] => {
+        return entries.map(async (entrie, index) => {
+            const delay = throttlingDelay * index;
+            await preloadImage(entrie.SourceURL, delay);
+
+            const img = new Image();
+            img.src = entrie.SourceURL;
+
+            return img;
+        });
+    };
+
+    const awaitPromiseImages = async (
+        promiseImages: Promise<HTMLImageElement>[]
+    ): Promise<HTMLImageElement[]> => {
+        const images: HTMLImageElement[] = [];
+
+        for (const promiseImage of promiseImages) {
+            const image = await promiseImage;
+            images.push(image);
+        }
+
+        return images;
+    };
+
+    const convertImagesToChapterImages = (
+        images: HTMLImageElement[]
+    ): ChapterImages[] => {
+        const chapterImages: ChapterImages[] = [];
+        let i = 0;
+
+        while (i < images.length) {
+            let nextSrc: string | undefined = undefined;
+
+            // should display two image
+            if (
+                $doublePage &&
+                images[i] &&
+                images[i].width < images[i].height &&
+                images[i + 1] &&
+                images[i + 1].width < images[i + 1].height
+            ) {
+                nextSrc = images[i + 1].src;
+            }
+
+            const chapterImage = {
+                src: images[i].src,
+                nextSrc,
+            };
+
+            chapterImages.push(chapterImage);
+
+            if (chapterImage.nextSrc) {
+                // because we already process the i+1 image
+                i += 2;
+
+                // make the currentImageIndex match the chapterImages data
+                if (currentImageIndex >= chapterImages.length) {
+                    currentImageIndex--;
+                }
+            } else {
+                i++;
+            }
+        }
+
+        return chapterImages;
+    };
+
+    // Convert URLs to a useful data structure
+    const getChapterImages = async (): Promise<ChapterImages[]> => {
+        const promiseImages = getPromiseImages(item.Entries);
+
+        // We await all promises to have an HTMLImageElement[] instead of Promise<HTMLImageElement>[]
+        const images = await awaitPromiseImages(promiseImages);
+
+        return convertImagesToChapterImages(images);
+    };
+
+    onMount(async () => {
+        setImageWidth();
+        chapterImages = await getChapterImages();
```

Comment:
no need to await. Just let chapterImages as a promise and use await within the template
{#await promise}
using await you can set a "loading" animation easily